src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback } from 'react';
+import Button from '@material-ui/core/Button';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FormControl from '@material-ui/core/FormControl';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { useForm } from 'react-final-form';
+
+const useStyles = makeStyles({
+  row: {
+    zIndex: 10,
+    margin: '1em 0',
+    backgroundColor: 'white',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  formControl: {
+    display: 'flex',
+    flexDirection: 'columns',
+    alignItems: 'strech',
+  },
+});
+
+const PopupFieldRow = ({ popupField, onChange }) => {
+  const classes = useStyles();
+
+  const onSuffixChange = useCallback(e => {
+    const newPopupField = { ...popupField };
+    newPopupField.suffix = e.target.value;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  const onPrefixChange = useCallback(e => {
+    const newPopupField = { ...popupField };
+    newPopupField.prefix = e.target.value;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  const onDefaultChange = useCallback(e => {
+    const newPopupField = { ...popupField };
+    newPopupField.default = e.target.value;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
+
+  const onDeletePopupField = useCallback(() => {
+    const newPopupField = { ...popupField };
+    newPopupField.deleted = true;
+    onChange(newPopupField);
+  }, [onChange, popupField]);
```

Comment:
This could maybe written more briefly.

Reply:
First, 

```js
e => {
    const newPopupField = { ...popupField };
    newPopupField.suffix = e.target.value;
    onChange(newPopupField);
  };

// Could be written like:
({ target: { value } }) => onChange({ ...popupField, suffix: value });


// But `suffix` may also be a given variable, so:
const fieldName = 'suffix';
({ target: { value } }) => onChange({ ...popupField, [fieldName]: value });
```

Then, all this block may be written something like:

```js
const setPopupFieldValue = (popupField, value, fieldName, callback) =>
  callback({ ...popupField, [fieldName]: value });

const onSuffixChange = useCallback(
  ({ target: { value } }) => setPopupFieldValue(popupField, 'suffix', value, onChange),
  [onChange, popupField],
);

const onPrefixChange = useCallback(
  ({ target: { value } }) => setPopupFieldValue(popupField, 'prefix', value, onChange),
  [onChange, popupField],
);

const onDefaultChange = useCallback(
  ({ target: { value } }) => setPopupFieldValue(popupField, 'default', value, onChange),
  [onChange, popupField],
);

const onDeletePopupField = useCallback(
  () => setPopupFieldValue(popupField, 'deleted', true, onChange),
  [onChange, popupField],
);

// This code block has NOT been tested.
```

Reply:
I didn't saw your suggestion, but I refactored it. Tell me what you think of it.